### PR TITLE
create python jdbc to bigquery template

### PR DESCRIPTION
**Pull request by sjlva**

### CODE REVIEW

**Comment by shashank-google** [comment]
@sjlva Can you please identify yourself starting with are you Google Employee or outside Google.
Also I want to understand, what encouraged you to create pyspark template for JDBC to BQ (instead of picking other open github issues) ?

**Comment by sjlva** [comment]
@shashank-google I work outside Google.
I'll pick some other github issues in future. But, ATM I needed an JDC to BQ template in python for an professional project. So, instead of creating my own templates I found this repo and thought it was a good idea to submit this template.

File: python/dataproc_templates/jdbc/README.md [hunk 1]
```diff
@@ -375,4 +375,102 @@ export JARS="gs://my-gcp-proj/jars/mysql-connector-java-8.0.29.jar,gs://my-gcp-p
...
+```
+
+## General execution: 
+
+```
+export GCP_PROJECT=<gcp-project-id> 
+export REGION=<region>  
+export GCS_STAGING_LOCATION=<gcs staging location> 
+export SUBNET=<subnet>   
+export JARS="<gcs_path_to_jdbc_jar_files>/mysql-connector-java-8.0.29.jar,<gcs_path_to_jdbc_jar_files>/postgresql-42.2.6.jar,<gcs_path_to_jdbc_jar_files>/mssql-jdbc-6.4.0.jre8.jar,<gcs_path_to_jdbc_jar_files>/spark-bigquery-latest_2.12.jar"
```
Comment (shashank-google) [inline python/dataproc_templates/jdbc/README.md:462]: Can BQ-Spark connector be directly referenced from its gcs location ?
gs://spark-lib/bigquery/spark-bigquery-latest_2.12.jar

Comment (shashank-google) [inline python/dataproc_templates/jdbc/README.md:462]: In the instructions only one jdbc jar (eith mysql, pg or sql server) should be ideally referenced.

File: python/dataproc_templates/jdbc/README.md [hunk 2]
```diff
@@ -375,4 +375,102 @@ export JARS="gs://my-gcp-proj/jars/mysql-connector-java-8.0.29.jar,gs://my-gcp-p
...
+--jdbc.bigquery.input.url="jdbc:mysql://<hostname>:<port>/<dbname>?user=<username>&password=<password>" \
+--jdbc.bigquery.input.driver="<jdbc-driver-class-name>" \
+--jdbc.bigquery.input.table="input table name or subquery with where clause filter" \
+--jdbc.bigquery.output.mode="<append|overwrite|ignore|errorifexists>" \
+--jdbc.bigquery.output.dataset="<bigquery-dataset-name>" \
+--jdbc.bigquery.output.table="<bigquery-dataset-table>" \
+--jdbc.bigquery.temp.bucket.name="<temp-bq-bucket-name>"
+
+```
+
```
Comment (shashank-google) [inline python/dataproc_templates/jdbc/README.md:476]: Missing example submission.

**Review by shashank-google — CHANGES_REQUESTED** [review]

**Comment by shashank-google** [comment]
I have looked at Readme files.
@NiloFreitas request you to please review the rest of the python code.

File: python/dataproc_templates/jdbc/README.md [hunk 3]
```diff
@@ -375,4 +375,102 @@ export JARS="gs://my-gcp-proj/jars/mysql-connector-java-8.0.29.jar,gs://my-gcp-p
 --jdbctogcs.output.mode="overwrite" \
 --jdbctogcs.output.format="csv" \
 --jdbctogcs.output.partitioncolumn="department_id"
-```
\ No newline at end of file
+```
```
Comment (shashank-google) [inline python/dataproc_templates/jdbc/README.md:378]: Missing corresponding changes in 
1. /Readme.md
2. /python/README.md

File: python/dataproc_templates/jdbc/README.md (hunk 1 above)
Comment (sjlva) [reply python/dataproc_templates/jdbc/README.md:462]: Yes, it can. Ref: https://cloud.google.com/dataproc/docs/tutorials/bigquery-connector-spark-example

Comment (sjlva) [reply python/dataproc_templates/jdbc/README.md:462]: Ok. I'll fix that. But It thought that referencing all jars were the norm. Like those examples: https://github.com/GoogleCloudPlatform/dataproc-templates/tree/main/python/dataproc_templates/jdbc

**Review by NiloFreitas — CHANGES_REQUESTED** [review]
Missing unit tests

File: python/test/jdbc/test_jdbc_to_bigquery.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        assert parsed_args["jdbc.bigquery.input.partitioncolumn"] == "column"
+        assert parsed_args["jdbc.bigquery.input.lowerbound"] == "1"
+        assert parsed_args["jdbc.bigquery.input.upperbound"] == "2"
+        assert parsed_args["jdbc.bigquery.numpartitions"] == "5"
+        assert parsed_args["jdbc.bigquery.output.mode"] == "append"  
+        assert parsed_args["jdbc.bigquery.output.dataset"] == "bq-dataset"  
+        assert parsed_args["jdbc.bigquery.output.table"] == "bq-table"  
+        assert parsed_args["jdbc.bigquery.temp.bucket.name"] == "bucket-name"  
+
+    def test_run_pass_args3(self):
```
Comment (NiloFreitas) [inline python/test/jdbc/test_jdbc_to_bigquery.py:59]: Fix method name to: test_run_pass_args2

**Comment by NiloFreitas** [comment]
Hi! Could you try to rebase with main and fix the conflicts in the README and in the template_constants.py file?

**Comment by sjlva** [comment]
@NiloFreitas Just rebased with main, fixing conflicts in README and template_constants.py.

Also, fixed the method name to test_run_pass_args2

**Review by NiloFreitas — APPROVED** [review]
LGTM! Nice work!

**Review by shashank-google — APPROVED** [review]

**Pull request merged**